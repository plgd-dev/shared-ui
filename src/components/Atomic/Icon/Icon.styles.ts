import { css } from '@emotion/react'

export const icon = css`
    display: flex;
    align-items: center;
    justify-content: center;
`
