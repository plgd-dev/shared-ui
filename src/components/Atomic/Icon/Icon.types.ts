import { FC, SVGProps } from 'react'

export type Props = {
    className?: string
    icon: string
    id?: string
    onCompleted?: (name: string, SvgIcon: FC<SVGProps<SVGSVGElement>> | undefined) => void
    onError?: (err: Error) => void
    size?: number
} & SVGProps<SVGSVGElement>

export const defaultProps = {
    size: 16,
}
