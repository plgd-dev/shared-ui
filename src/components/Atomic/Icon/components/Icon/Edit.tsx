// Auto-generated file created by svgr-cli source svg-template.js
// Run yarn icons:create to update
// Do not edit
import * as React from 'react'
import { SVGProps } from 'react'
const SvgEdit = (props: SVGProps<SVGSVGElement>) => (
    <svg width={props.width || 16} height={props.height || 16} viewBox='0 0 20 20' fill='none' xmlns='http://www.w3.org/2000/svg' {...props}>
        <path
            d='M17.178 2.207a1.84 1.84 0 0 0-2.605 0l-.99.987 3.22 3.22.99-.99a1.84 1.84 0 0 0 0-2.605l-.615-.612Zm-9.84 7.236c-.202.2-.356.447-.445.72l-.974 2.92a.793.793 0 0 0 .191.81c.21.214.523.286.81.19l2.92-.973c.27-.092.517-.243.72-.444l5.503-5.506-3.223-3.223-5.503 5.506ZM4.823 3.598a3.158 3.158 0 0 0-3.157 3.158v8.42a3.158 3.158 0 0 0 3.157 3.157h8.42a3.158 3.158 0 0 0 3.158-3.157v-3.158a1.051 1.051 0 1 0-2.105 0v3.158c0 .582-.47 1.052-1.053 1.052h-8.42c-.582 0-1.052-.47-1.052-1.052v-8.42c0-.582.47-1.053 1.052-1.053h3.158a1.051 1.051 0 1 0 0-2.105H4.824Z'
            fill='#81868C'
        />
    </svg>
)
export default SvgEdit
