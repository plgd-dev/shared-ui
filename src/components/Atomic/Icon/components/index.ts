export { default as IconActions } from './IconActions'
export { default as IconArrowDetail } from './IconArrowDetail'
export { default as IconArrowDownNoPadding } from './IconArrowDownNoPadding'
export { default as IconArrowDown } from './IconArrowDown'
export { default as IconArrowLeft } from './IconArrowLeft'
export { default as IconArrowRight } from './IconArrowRight'
export { default as IconArrowTriangleFullUp } from './IconArrowTriangleFullUp'
export { default as IconBell } from './IconBell'
export { default as IconCertificate } from './IconCertificate'
export { default as IconChat } from './IconChat'
export { default as IconCloseCircle } from './IconCloseCircle'
export { default as IconClose } from './IconClose'
export { default as IconCloudError } from './IconCloudError'
export { default as IconCloudSuccess } from './IconCloudSuccess'
export { default as IconCloudWarning } from './IconCloudWarning'
export { default as IconCollapse } from './IconCollapse'
export { default as IconCopy } from './IconCopy'
export { default as IconDashboard } from './IconDashboard'
export { default as IconDeviceUpdate } from './IconDeviceUpdate'
export { default as IconDevices } from './IconDevices'
export { default as IconDocs } from './IconDocs'
export { default as IconDownload } from './IconDownload'
export { default as IconEdit } from './IconEdit'
export { default as IconError } from './IconError'
export { default as IconFilePem } from './IconFilePem'
export { default as IconFileUpload } from './IconFileUpload'
export { default as IconGlobe } from './IconGlobe'
export { default as IconHidePassword } from './IconHidePassword'
export { default as IconInfo } from './IconInfo'
export { default as IconIntegrations } from './IconIntegrations'
export { default as IconLink } from './IconLink'
export { default as IconLoader } from './IconLoader'
export { default as IconLock } from './IconLock'
export { default as IconLog } from './IconLog'
export { default as IconLogout } from './IconLogout'
export { default as IconNet } from './IconNet'
export { default as IconNetwork } from './IconNetwork'
export { default as IconNotification } from './IconNotification'
export { default as IconPendingCommands } from './IconPendingCommands'
export { default as IconPlus } from './IconPlus'
export { default as IconRefresh } from './IconRefresh'
export { default as IconRemoteClients } from './IconRemoteClients'
export { default as IconSearch } from './IconSearch'
export { default as IconSettings } from './IconSettings'
export { default as IconShield } from './IconShield'
export { default as IconShowPassword } from './IconShowPassword'
export { default as IconSortDown } from './IconSortDown'
export { default as IconSortUp } from './IconSortUp'
export { default as IconSuccess } from './IconSuccess'
export { default as IconTableArrowDown } from './IconTableArrowDown'
export { default as IconTableArrowUp } from './IconTableArrowUp'
export { default as IconToastClose } from './IconToastClose'
export { default as IconTrash } from './IconTrash'
export { default as IconWarning } from './IconWarning'
