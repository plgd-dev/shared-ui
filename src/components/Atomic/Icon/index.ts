export { default } from './Icon'
export * from './Icon'

export * from './components'

export * from './utils'
