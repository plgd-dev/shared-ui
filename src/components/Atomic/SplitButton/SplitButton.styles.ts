import { css } from '@emotion/react'
import { colors } from '../_utils/colors'
import { ThemeType, get } from '../_theme'

export const splitButton = css`
    display: flex;
`

export const leftButton = css`
    border-top-right-radius: 0;
    border-bottom-right-radius: 0;
`

export const rightButton = css`
    border-top-left-radius: 0;
    border-bottom-left-radius: 0;
    border-left: 0;
    padding: 14px;
`

export const floatingMenu = (theme: ThemeType) => css`
    z-index: 10;
    background: ${get(theme, `SplitButton.floatingMenu.background`)};
    border: 1px solid ${get(theme, `SplitButton.floatingMenu.borderColor`)};
    box-shadow: 0 30px 40px rgba(28, 52, 99, 0.1);
    border-radius: 8px;
    padding: 10px 20px;
`

export const item = css`
    display: flex;
    align-items: center;
    text-decoration: none;
    color: ${colors.neutral500};
    transition: all 0.3s;
    margin: 6px 0;
    padding: 4px 0;
    cursor: pointer;

    &:hover {
        color: ${colors.primary};
    }
`

export const itemIcon = css`
    margin-right: 10px;
`

export const itemLabel = css`
    font-size: 14px;
    line-height: 14px;
`
