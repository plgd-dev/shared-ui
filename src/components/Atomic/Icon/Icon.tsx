import { FC, forwardRef, SVGProps, useEffect, useRef, useState } from 'react'
import { Props, defaultProps } from './Icon.types'
import * as styles from './Icon.styles'

export const Icon = forwardRef<any, Props>((props, ref) => {
    const { className, icon, id, size, onError, onCompleted, ...rest } = { ...defaultProps, ...props }

    const [loading, setLoading] = useState(false)
    const [iconLoad, setIconLoad] = useState('')
    const [error, setError] = useState<Error | undefined>(undefined)
    const ImportedIconRef = useRef<FC<SVGProps<SVGSVGElement>> | undefined>(undefined)

    const IconComponent = ImportedIconRef.current ? ImportedIconRef.current : () => null

    const parseComponentName = (string: string) =>
        string
            .split('-')
            .map((part) => part.charAt(0).toUpperCase() + part.slice(1))
            .join('')

    useEffect(() => {
        setLoading(true)
        const componentName = parseComponentName(icon)
        const importIcon = async (): Promise<void> => {
            try {
                await import(`./components/${componentName}.tsx`).then((IconComponent) => {
                    ImportedIconRef.current = IconComponent.default
                    return IconComponent.default
                })
                onCompleted?.(icon, ImportedIconRef.current)
            } catch (err: any) {
                onError?.(err)
                setError(err)
            } finally {
                setLoading(false)
            }
        }
        importIcon().then(() => {
            setIconLoad(icon)
        })
    }, [icon, onCompleted, onError])

    if (error) {
        console.error(error)
        return null
    }

    return (
        <div
            className={className}
            css={styles.icon}
            data-icon={iconLoad}
            id={id}
            ref={ref}
            style={{
                width: size,
                height: size,
            }}
        >
            {loading ? null : <IconComponent {...rest} height={size} width={size} />}
        </div>
    )
})

Icon.displayName = 'Icon'

export default Icon
