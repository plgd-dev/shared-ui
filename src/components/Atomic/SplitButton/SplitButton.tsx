import { cloneElement, FC, ReactElement, useEffect, useRef, useState } from 'react'
import Button from '../Button'
import { Props, defaultProps, SplitButtonItemType } from './SplitButton.types'
import * as styles from './SplitButton.styles'
import Icon, { convertSize, IconArrowDown } from '../Icon'
import { offset, shift, useFloating } from '@floating-ui/react'
import isFunction from 'lodash/isFunction'
import { motion } from 'framer-motion'

const SplitButton: FC<Props> = (props) => {
    const { defaultOpen, children, variant, className, menuProps, items, disabled, loading, onClick, dataTestId, dataTestIdDropdown, ...rest } = {
        ...defaultProps,
        ...props,
    }
    const [open, setOpen] = useState(defaultOpen)
    const ref = useRef(null)
    const { x, y, refs, strategy } = useFloating({
        placement: menuProps?.placement || 'bottom-end',
        strategy: 'fixed',
        middleware: [shift(), offset(4)],
    })

    useEffect(() => {
        function handleClickOutside(event: any) {
            // @ts-ignore
            if (ref?.current && !ref?.current?.contains(event.target)) {
                setOpen(false)
            }
        }
        if (open) {
            document.addEventListener('mousedown', handleClickOutside)
            return () => {
                document.removeEventListener('mousedown', handleClickOutside)
            }
        }
    }, [ref, open])

    const getIcon = (item: SplitButtonItemType) => {
        if (item.icon) {
            return typeof item.icon === 'string' ? (
                <Icon css={styles.itemIcon} icon={item.icon} size={20} />
            ) : (
                cloneElement(item.icon as ReactElement, { ...convertSize(20), css: styles.itemIcon })
            )
        }

        return null
    }

    return (
        <div ref={ref}>
            <div css={styles.splitButton} ref={refs.setReference}>
                <Button
                    {...rest}
                    css={styles.leftButton}
                    dataTestId={dataTestId}
                    disabled={disabled}
                    loading={loading}
                    onClick={(e) => {
                        open && setOpen(false)
                        isFunction(onClick) && onClick(e)
                    }}
                    variant={variant}
                >
                    {children}
                </Button>
                <Button
                    css={styles.rightButton}
                    data-test-id={dataTestIdDropdown}
                    disabled={disabled}
                    onClick={() => !loading && setOpen(!open)}
                    variant={variant}
                >
                    <motion.div
                        animate={{
                            rotate: open ? 180 : 0,
                        }}
                        transition={{
                            duration: 0.3,
                            damping: 25,
                            stiffness: 500,
                        }}
                    >
                        <IconArrowDown />
                    </motion.div>
                </Button>

                {open && (
                    <div
                        css={styles.floatingMenu}
                        ref={refs.setFloating}
                        style={{
                            position: strategy,
                            top: y ?? 0,
                            left: x ?? 0,
                            width: 'max-content',
                        }}
                    >
                        {items
                            .filter((item) => !item.hidden)
                            .map((item) => {
                                return (
                                    item.component || (
                                        <div
                                            css={styles.item}
                                            key={item.id || item.label}
                                            onClick={(e) => {
                                                open && setOpen(false)
                                                isFunction(item.onClick) && item.onClick(e)
                                            }}
                                        >
                                            {getIcon(item)}
                                            <span css={styles.itemLabel}>{item.label}</span>
                                        </div>
                                    )
                                )
                            })}
                    </div>
                )}
            </div>
        </div>
    )
}

SplitButton.defaultProps = defaultProps
SplitButton.displayName = 'SplitButton'

export default SplitButton
