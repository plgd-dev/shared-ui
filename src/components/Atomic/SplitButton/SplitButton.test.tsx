import React from 'react'
import { render } from '@testing-library/react'
import SplitButton from './SplitButton'
import { IconPlus } from '../Icon'

describe('<SimpleStripTable>', () => {
    it('render correctly - snapshot', () => {
        const items = [
            {
                onClick: () => console.log('onItemClick'),
                label: 'Item 1',
                icon: 'lock',
            },
            {
                onClick: () => console.log('onItemClick'),
                label: 'Item 2',
                icon: 'log',
            },
            {
                onClick: () => console.log('onItemClick'),
                label: 'Item 3',
                icon: 'net',
            },
        ]
        const { asFragment } = render(
            <div>
                <SplitButton children='Primary' items={items} variant='primary' />
                <SplitButton children='Secondary' items={items} variant='secondary' />
                <SplitButton children='Primary' disabled={true} items={items} variant='primary' />
                <SplitButton children='Secondary' disabled={true} items={items} variant='secondary' />
                <SplitButton children='Primary' icon={<IconPlus />} items={items} variant='primary' />
                <SplitButton children='Secondary' icon={<IconPlus />} items={items} variant='secondary' />
                <SplitButton children='Primary' items={items} loading={true} variant='primary' />
                <SplitButton children='Secondary' items={items} loading={true} variant='secondary' />
            </div>
        )

        expect(asFragment()).toMatchSnapshot()
    })
})
