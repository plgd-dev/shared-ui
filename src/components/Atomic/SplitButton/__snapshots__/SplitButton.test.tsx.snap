// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SimpleStripTable> render correctly - snapshot 1`] = `
<DocumentFragment>
  @keyframes animation-0 {
  0% {
    -webkit-transform: rotate(0deg);
    -moz-transform: rotate(0deg);
    -ms-transform: rotate(0deg);
    transform: rotate(0deg);
  }

  50% {
    -webkit-transform: rotate(180deg);
    -moz-transform: rotate(180deg);
    -ms-transform: rotate(180deg);
    transform: rotate(180deg);
  }

  100% {
    -webkit-transform: rotate(360deg);
    -moz-transform: rotate(360deg);
    -ms-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-transition: all 0.25s;
  transition: all 0.25s;
  font-family: 'Poppins',sans-serif;
  font-style: normal;
  font-weight: 600;
  text-align: center;
  border-radius: 8px;
  white-space: nowrap;
  box-sizing: border-box;
  cursor: pointer;
  border: 1px solid;
  padding: 10px 24px;
  font-size: 14px;
  line-height: 16px;
  height: 48px;
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
}

.emotion-2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-transition: all 0.25s;
  transition: all 0.25s;
  font-family: 'Poppins',sans-serif;
  font-style: normal;
  font-weight: 600;
  text-align: center;
  border-radius: 8px;
  white-space: nowrap;
  box-sizing: border-box;
  cursor: pointer;
  border: 1px solid;
  padding: 10px 24px;
  font-size: 14px;
  line-height: 16px;
  height: 48px;
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
  border-left: 0;
  padding: 14px;
}

.emotion-7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-transition: all 0.25s;
  transition: all 0.25s;
  font-family: 'Poppins',sans-serif;
  font-style: normal;
  font-weight: 600;
  text-align: center;
  border-radius: 8px;
  white-space: nowrap;
  box-sizing: border-box;
  cursor: pointer;
  border: 1px solid;
  border: 1px solid;
  cursor: not-allowed!important;
  padding: 10px 24px;
  font-size: 14px;
  line-height: 16px;
  height: 48px;
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
}

.emotion-8 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-transition: all 0.25s;
  transition: all 0.25s;
  font-family: 'Poppins',sans-serif;
  font-style: normal;
  font-weight: 600;
  text-align: center;
  border-radius: 8px;
  white-space: nowrap;
  box-sizing: border-box;
  cursor: pointer;
  border: 1px solid;
  border: 1px solid;
  cursor: not-allowed!important;
  padding: 10px 24px;
  font-size: 14px;
  line-height: 16px;
  height: 48px;
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
  border-left: 0;
  padding: 14px;
}

.emotion-14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 8px;
}

.emotion-22 {
  width: 20px;
  height: 20px;
  overflow: hidden;
  display: block;
  -webkit-animation: 0.75s linear 0s infinite normal both running animation-0;
  animation: 0.75s linear 0s infinite normal both running animation-0;
  color: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 8px;
}

.emotion-26 {
  width: 20px;
  height: 20px;
  overflow: hidden;
  display: block;
  -webkit-animation: 0.75s linear 0s infinite normal both running animation-0;
  animation: 0.75s linear 0s infinite normal both running animation-0;
  color: #2261AE;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 8px;
}

<div>
    <div>
      <div
        class="emotion-0"
      >
        <button
          class="emotion-1"
          tabindex="1"
          type="submit"
        >
          Primary
        </button>
        <button
          class="emotion-2"
          tabindex="1"
          type="submit"
        >
          <div
            style="transform: none;"
          >
            <svg
              fill="none"
              height="16"
              viewBox="0 0 14 8"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1.707.293A1 1 0 0 0 .293 1.707l6 6a1 1 0 0 0 1.414 0l6-6A1 1 0 0 0 12.293.293L7 5.586z"
                fill="currentcolor"
              />
            </svg>
          </div>
        </button>
      </div>
    </div>
    <div>
      <div
        class="emotion-0"
      >
        <button
          class="emotion-1"
          tabindex="1"
          type="submit"
        >
          Secondary
        </button>
        <button
          class="emotion-2"
          tabindex="1"
          type="submit"
        >
          <div
            style="transform: none;"
          >
            <svg
              fill="none"
              height="16"
              viewBox="0 0 14 8"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1.707.293A1 1 0 0 0 .293 1.707l6 6a1 1 0 0 0 1.414 0l6-6A1 1 0 0 0 12.293.293L7 5.586z"
                fill="currentcolor"
              />
            </svg>
          </div>
        </button>
      </div>
    </div>
    <div>
      <div
        class="emotion-0"
      >
        <button
          class="emotion-7"
          tabindex="1"
          type="submit"
        >
          Primary
        </button>
        <button
          class="emotion-8"
          tabindex="1"
          type="submit"
        >
          <div
            style="transform: none;"
          >
            <svg
              fill="none"
              height="16"
              viewBox="0 0 14 8"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1.707.293A1 1 0 0 0 .293 1.707l6 6a1 1 0 0 0 1.414 0l6-6A1 1 0 0 0 12.293.293L7 5.586z"
                fill="currentcolor"
              />
            </svg>
          </div>
        </button>
      </div>
    </div>
    <div>
      <div
        class="emotion-0"
      >
        <button
          class="emotion-7"
          tabindex="1"
          type="submit"
        >
          Secondary
        </button>
        <button
          class="emotion-8"
          tabindex="1"
          type="submit"
        >
          <div
            style="transform: none;"
          >
            <svg
              fill="none"
              height="16"
              viewBox="0 0 14 8"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1.707.293A1 1 0 0 0 .293 1.707l6 6a1 1 0 0 0 1.414 0l6-6A1 1 0 0 0 12.293.293L7 5.586z"
                fill="currentcolor"
              />
            </svg>
          </div>
        </button>
      </div>
    </div>
    <div>
      <div
        class="emotion-0"
      >
        <button
          class="emotion-1"
          tabindex="1"
          type="submit"
        >
          <span
            class="emotion-14"
          >
            <svg
              fill="none"
              height="16"
              viewBox="0 0 20 20"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10 18.333A8.333 8.333 0 0 0 18.333 10 8.333 8.333 0 0 0 10 1.667 8.333 8.333 0 0 0 1.667 10 8.333 8.333 0 0 0 10 18.333m-.781-5.468V10.78H7.135a.78.78 0 0 1-.78-.78.78.78 0 0 1 .78-.782H9.22V7.136A.78.78 0 0 1 10 6.354a.78.78 0 0 1 .781.782v2.083h2.084a.78.78 0 0 1 .78.781.78.78 0 0 1-.78.781H10.78v2.084a.78.78 0 0 1-.781.78.78.78 0 0 1-.781-.78"
                fill="currentcolor"
              />
            </svg>
          </span>
          Primary
        </button>
        <button
          class="emotion-2"
          tabindex="1"
          type="submit"
        >
          <div
            style="transform: none;"
          >
            <svg
              fill="none"
              height="16"
              viewBox="0 0 14 8"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1.707.293A1 1 0 0 0 .293 1.707l6 6a1 1 0 0 0 1.414 0l6-6A1 1 0 0 0 12.293.293L7 5.586z"
                fill="currentcolor"
              />
            </svg>
          </div>
        </button>
      </div>
    </div>
    <div>
      <div
        class="emotion-0"
      >
        <button
          class="emotion-1"
          tabindex="1"
          type="submit"
        >
          <span
            class="emotion-14"
          >
            <svg
              fill="none"
              height="16"
              viewBox="0 0 20 20"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10 18.333A8.333 8.333 0 0 0 18.333 10 8.333 8.333 0 0 0 10 1.667 8.333 8.333 0 0 0 1.667 10 8.333 8.333 0 0 0 10 18.333m-.781-5.468V10.78H7.135a.78.78 0 0 1-.78-.78.78.78 0 0 1 .78-.782H9.22V7.136A.78.78 0 0 1 10 6.354a.78.78 0 0 1 .781.782v2.083h2.084a.78.78 0 0 1 .78.781.78.78 0 0 1-.78.781H10.78v2.084a.78.78 0 0 1-.781.78.78.78 0 0 1-.781-.78"
                fill="currentcolor"
              />
            </svg>
          </span>
          Secondary
        </button>
        <button
          class="emotion-2"
          tabindex="1"
          type="submit"
        >
          <div
            style="transform: none;"
          >
            <svg
              fill="none"
              height="16"
              viewBox="0 0 14 8"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1.707.293A1 1 0 0 0 .293 1.707l6 6a1 1 0 0 0 1.414 0l6-6A1 1 0 0 0 12.293.293L7 5.586z"
                fill="currentcolor"
              />
            </svg>
          </div>
        </button>
      </div>
    </div>
    <div>
      <div
        class="emotion-0"
      >
        <button
          class="emotion-1"
          tabindex="1"
          type="submit"
        >
          <span
            class="emotion-22"
          >
            <svg
              fill="none"
              height="20"
              size="20"
              viewBox="0 0 20 19"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18.575 10c.787 0 1.435-.641 1.324-1.42A10 10 0 0 0 9.682.005a10 10 0 0 0-5.173 18.353c.658.432 1.514.117 1.848-.595.335-.713.016-1.551-.616-2.02a7.15 7.15 0 0 1-2.114-8.985A7.15 7.15 0 0 1 17.01 8.584c.155.772.779 1.416 1.566 1.416"
                fill="currentcolor"
              />
            </svg>
          </span>
          Primary
        </button>
        <button
          class="emotion-2"
          tabindex="1"
          type="submit"
        >
          <div
            style="transform: none;"
          >
            <svg
              fill="none"
              height="16"
              viewBox="0 0 14 8"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1.707.293A1 1 0 0 0 .293 1.707l6 6a1 1 0 0 0 1.414 0l6-6A1 1 0 0 0 12.293.293L7 5.586z"
                fill="currentcolor"
              />
            </svg>
          </div>
        </button>
      </div>
    </div>
    <div>
      <div
        class="emotion-0"
      >
        <button
          class="emotion-1"
          tabindex="1"
          type="submit"
        >
          <span
            class="emotion-26"
          >
            <svg
              fill="none"
              height="20"
              size="20"
              viewBox="0 0 20 19"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18.575 10c.787 0 1.435-.641 1.324-1.42A10 10 0 0 0 9.682.005a10 10 0 0 0-5.173 18.353c.658.432 1.514.117 1.848-.595.335-.713.016-1.551-.616-2.02a7.15 7.15 0 0 1-2.114-8.985A7.15 7.15 0 0 1 17.01 8.584c.155.772.779 1.416 1.566 1.416"
                fill="currentcolor"
              />
            </svg>
          </span>
          Secondary
        </button>
        <button
          class="emotion-2"
          tabindex="1"
          type="submit"
        >
          <div
            style="transform: none;"
          >
            <svg
              fill="none"
              height="16"
              viewBox="0 0 14 8"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1.707.293A1 1 0 0 0 .293 1.707l6 6a1 1 0 0 0 1.414 0l6-6A1 1 0 0 0 12.293.293L7 5.586z"
                fill="currentcolor"
              />
            </svg>
          </div>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
